tools/syz-declextract/syz-declextract.cpp

```diff
@@ -64,6 +106,230 @@ bool beginsWith(const std::string_view &str, const std::string_view begin) {
 
 bool contains(const std::string_view &str, const std::string_view sub) { return str.find(sub) != std::string::npos; }
 
+bool isPadding(const std::string &fieldName) {
+  return contains(fieldName, "pad") || contains(fieldName, "unused") || contains(fieldName, "_reserved");
+}
+
+const std::string makeArray(const std::string &type, const size_t len = 0) {
+  if (len == 1) {
+    return type;
+  }
+  if (len) {
+    return "array[" + type + ", " + std::to_string(len) + "]";
+  }
+  return "array[" + type + "]";
+}
+
+const std::string makeArray(const std::string &type, const size_t min, const size_t max) {
+  if (min == max) {
+    return makeArray(type, min);
+  }
+  return "array[" + type + ", " + std::to_string(min) + ":" + std::to_string(max) + "]";
+}
+
+const std::string makePtr(const std::string &dir, const std::string &type, bool isOpt = false) {
+  if (type == "void") {
```

Comment:
The string case and "void" should be handled in the same place. Either both here, or both where you handle "string". They are effectively the same (we special-case some pointer types), but for some reason string is in one place, while void is in another place.